src/backend/distributed/executor/intermediate_results.c

```diff
@@ -848,9 +847,6 @@ fetch_intermediate_results(PG_FUNCTION_ARGS)
 		totalBytesWritten += FetchRemoteIntermediateResult(connection, resultId);
 	}
 
-	RemoteTransactionCommit(connection);
```

Comment:
What prompted this change?

Reply:
In `fetch_intermediate_results()` we were not forcing new connections, so we could be using remote connections which had already executed some commands. Committing and closing connections would be wrong in that case. For example,

```sql
BEGIN;
UPDATE dist_table a = a + 1;
SELECT fetch_intermediate_results(...);
DELETE FROM dist_table;
END;
```

wouldn't work, because `fetch_intermediate_results()` would have committed and closed the connections and connections weren't open to send `DELETE`s.

Also commit and close would delete remote intermediate files, since we delete intermediate files at commit.